### PR TITLE
Zillow XML Feed

### LINKED ISSUE #31: Generate Builder Digital Experience - BDX Output (XML Generation)
The current system can generate an XML file that can be used as input for 3rd party websites to list the homes from hubblehomes.com

There is an XML specification that can be referenced

image

### PR DESCRIPTION
**Pull request by varunmitra**
Fix #31 

Test URLs:
- Before: https://main--hubblehomes-com--aemsites.hlx.live/
- After: https://issue-31--hubblehomes-com--aemsites.hlx.live/

### CODE REVIEW

File: .github/xml-feed/generate-xml-feed.js [hunk 1]
```diff
@@ -0,0 +1,310 @@
+const { XMLParser, XMLBuilder, XMLValidator } = require('fast-xml-parser');
+const fs = require('fs');
```
Comment (codevoyage) [inline .github/xml-feed/generate-xml-feed.js:2]: XML Validator is not being used anywhere?

File: .github/xml-feed/generate-xml-feed.js [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+    <CorporateState>${entry['CorporateState']}</CorporateState>
+    <CorporateName>${entry['CorporateName']}</CorporateName>
+      <Builder>
+        <BuilderNumber>${entry['BuilderNumber']}</BuilderNumber>
+        <BrandName>${entry['BrandName']}</BrandName>
+        <ReportingName>${entry['ReportingName']}</ReportingName>
+        <DefaultLeadsEmail LeadsPerMessage="1">${entry['DefaultLeadsEmail']}</DefaultLeadsEmail>
+        <BuilderWebsite>${entry['BuilderWebsite']}</BuilderWebsite>`;
+  });
+  const subdivisionData = await fetchWorkbook([Sheets.SUBDIVISION, Sheets.SALES_OFFICES]);
```
Comment (codevoyage) [inline .github/xml-feed/generate-xml-feed.js:283]: What happens if the fetchWorkbook calls fail?

File: .github/xml-feed/generate-xml-feed.js [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+  }
+  throw new Error('Failed to fetch workbook');
+}
+
+async function addInventoryData(model, community) {
+  const inventoryData = await fetchWorkbook(Sheets.INVENTORY);
+  let inventoryXml = '';
+  for (const inventoryhome of inventoryData.data) {
+    if (inventoryhome.community.toLowerCase() === community.toLowerCase() && inventoryhome['model name'].toLowerCase() === model.toLowerCase()) {
+      inventoryXml += ` <Spec Type="SingleFamily">
```
Comment (codevoyage) [inline .github/xml-feed/generate-xml-feed.js:32]: If any text data inserted into the XML (e.g., description, Caption, etc.) contains characters like &, <, or >, could it break the XML structure.? Do the special characters need to be escaped?

File: .github/xml-feed/generate-xml-feed.js (hunk 2 above)
Comment (varunmitra) [reply .github/xml-feed/generate-xml-feed.js:283]: if the call fails, we will see an error in the Github workflow console. error will looks like: 
image

Comment (varunmitra) [reply .github/xml-feed/generate-xml-feed.js:283]: you can also see the trace here: https://github.com/aemsites/hubblehomes-com/actions/runs/10271678159/job/28422206951#step:6:1

**Review by samuelopra — APPROVED** [review]
lgtm!

**Pull request merged**